sdk/file_sdk/Common/tag.cpp

```diff
@@ -81,6 +81,17 @@ vector<string> Tag::GetAllIds(const vector<pair<string, string>>& properties) {
   return labelsIds;
 }
 
+void Tag::AddExtendedProperty(ExtendedProperty extendedProperty) {
+  auto it = std::find_if(mExtendedProperties.cbegin(), mExtendedProperties.cend(), [&extendedProperty](const ExtendedProperty& p) {

```

Comment:
you can move this code as a static helper in ExtendedProperty class and reuse it (not a must)